### PR TITLE
fix(firewall): make zones-v2 say when Zone-Based Firewall is off

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** on a controller without ZBF configured,
`list_firewall_zones_v2` returns a bare `[]`. An empty list from this
tool reads as "no zones exist", which is not a state a ZBF-enabled site
can be in — the built-in zones always exist once the feature is on.

**Context:** the integration zones route reports an explicit 400
(`api.firewall.zone-based-firewall-not-configured`) for the same
condition; the v2 route just returns empty.

**Fix:** raise a clear `APIError` for the empty case — "Zone-Based
Firewall does not appear to be configured on this site" — pointing at the
legacy firewall tools, instead of an empty list that reads as an answer.

**Live verification (Network 10.5.67):** verified against a site running
the legacy ruleset.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

ZBF-enabled sites are unaffected; only the impossible empty state changes
behavior.

## Testing

- Tests cover the empty-list error path (message names the legacy tools)
  and the populated-zones passthrough.
- `pytest tests/unit/` passes; pre-commit hooks pass for all files this
  PR touches.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; empty-state behavior verified
against a live Network 10.5.67 controller running the legacy ruleset.
**Security Review:** Completed — all fixture values are synthetic.

### CODE REVIEW

File: tests/unit/tools/test_firewall_policies.py [hunk 1]
```diff
@@ -1585,19 +1585,30 @@ async def test_list_firewall_zones_v2_returns_id_mapping(
...
+            with patch("src.tools.firewall_policies.UniFiClient") as MockClient:
+                mock_client = AsyncMock()
+                MockClient.return_value.__aenter__.return_value = mock_client
+                mock_client.is_authenticated = True
+                mock_client.get.return_value = empty_response
 
-            result = await list_firewall_zones_v2("default", local_settings)
-            assert result == []
+                with pytest.raises(APIError, match="Zone-Based Firewall"):
+                    await list_firewall_zones_v2("default", local_settings)
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply tests/unit/tools/test_firewall_policies.py:1611]: Fixed in 5b87808 — the test now asserts the legacy-tools guidance and the site scoping survive, not just the feature name.

File: src/tools/firewall_policies.py [hunk 2]
```diff
@@ -331,6 +331,21 @@ async def list_firewall_zones_v2(
...
+        # zones route reports as an explicit 400
+        # (api.firewall.zone-based-firewall-not-configured). Say so, rather
+        # than returning [] that reads as "no zones exist".
+        if not zones:
+            raise APIError(
+                "The v2 firewall zone route returned no zones. Zone-Based "
+                "Firewall does not appear to be configured on this "
+                "controller; a ZBF-enabled site always reports its built-in "
+                "zones. Legacy firewall rules are available via "
+                "list_firewall_rules."
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/firewall_policies.py]: Fixed in 5b87808: the message now says site.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.

**Comment by enuno** [comment]
Summary: request changes

Critical findings:
- Merge conflict / dirty merge state blocks safe review.

Suggestions:
1. Rebase onto main and resolve conflicts in: CHANGELOG.md, src/tools/firewall_policies.py, tests/unit/tools/test_firewall_policies.py.
2. Re-run the touched test suite and pre-commit.
3. Re-check the feature behavior called out in the PR description.

Verification checklist:
- [ ] Rebasing completed cleanly
- [ ] Tests pass
- [ ] Lint/type checks pass
- [ ] Documentation remains aligned with implementation

**Review by enuno — CHANGES_REQUESTED** [review]
Automated review: merge conflicts/dirtiness detected (CONFLICTING/DIRTY). Please rebase and resolve before resubmitting.

**Comment by jpennin5** [comment]
Rebased onto current `main` (f33e01b) — the conflict was in CHANGELOG, and it is resolved.

Resolution was gated rather than eyeballed: every CHANGELOG bullet already on `main` is verified still present (union, not overwrite), the sibling work merged into the shared files is verified intact, the full unit suite passes, and pre-commit passes for every file the PR touches.

Codecov reports all modified lines covered.

GitHub now reports this PR mergeable.

**Comment by enuno** [comment]
## Summary
Request changes.

## Critical findings
- CI is not green on this PR.
- Failing checks: Mermaid Diagram Sync Assistant; Pre-commit Hooks.
- Touched scope: CHANGELOG.md, src/tools, tests/unit.

## Suggestions
1. Fix the failing checks on the current branch and rerun them until the gate is green.
2. Re-verify the changed code paths and tests for the touched scope above.
3. If any failing security or lint checks are expected, document the rationale and follow-up needed.

## Verification checklist
- [ ] `pre-commit run --all-files` passes
- [ ] Security/lint checks are green or explicitly justified
- [ ] Relevant unit tests cover the changed behavior
- [ ] No unintended API or documentation regressions

**Review by enuno — CHANGES_REQUESTED** [review]
CI is red on this PR (Mermaid Diagram Sync Assistant; Pre-commit Hooks). Please resolve the failing checks before merge.
